delivery.go

```diff
@@ -58,6 +62,21 @@ type Delivery struct {
 	Body []byte
 }
 
+// Span returns context and a span that for the delivery
+// the resulting span is linked to the publication that created it, if it has
+// the appropraite headers set. See [context-propagation] for more details
+//
+// [context-propagation]: https://opentelemetry.io/docs/concepts/context-propagation/
+func (d *Delivery) Span(ctx context.Context, options ...trace.SpanStartOption) (context.Context, trace.Span) {
```

Comment:
this is an okay route-- clients can use their own context + span to indicate boundaries of a batch, and then get child spans for each delivery, with each span linked to the publication.

I also provide access to the Link for a delivery in case they want to combine multiple links into one span for their batch (that's what I would prefer in my use case, but that's because I'm defining a naturally batching consumer).

The tradeoff is that without storing additional state, we're relying on the client to tell us the context when we go to ack nack, which could lead to errors.

If we instead store the span on the delivery, we can close it when we ack, after inserting a child `settle` span for the ack itself. This has the implication that every consumer needs to `settle` the delivery even if their in autoack mode in order for them to see spans in their telemetry info (spans are usually not sent until they are closed). In `autoack` mode the `settle` method would just close the span, with no further implications at the wire level.

Reply:
> If we instead store the span on the delivery, we can close it when we ack, after inserting a child settle span for the ack itself

I prefer this approach, TBH. Relying on the user would make this implementation brittle. I'm ok with having a limitation handling `autoack`, because that's really not a recommended practice. `autoack` is synonym of `YOLO I don't care about my data, just GO!`